Server/mods/deathmatch/logic/CMainConfig.cpp

```diff
@@ -855,44 +855,53 @@ bool CMainConfig::AddMissingSettings()
     if (!g_pGame->IsUsingMtaServerConf())
         return false;
 
-    SString strTemplateFilename = PathJoin(g_pServerInterface->GetServerModPath(), "mtaserver.conf.template");
+    const std::string templateFileName = PathJoin(g_pServerInterface->GetServerModPath(), SETTINGS_TEMPLATE_PATH);
+    if (!FileExists(templateFileName))
+        return false;
 
-    if (!FileExists(strTemplateFilename))
+    std::unique_ptr<CXMLFile> templateFile(g_pServerInterface->GetXML()->CreateXML(templateFileName.c_str()));
+    if (!templateFile || !templateFile->Parse())
+    {
+        CLogger::ErrorPrintf("Failed to parse template file: '%s'\n", templateFileName.c_str());
         return false;
+    }
 
-    CXMLFile* pFileTemplate = g_pServerInterface->GetXML()->CreateXML(strTemplateFilename);
-    CXMLNode* pRootNodeTemplate = pFileTemplate && pFileTemplate->Parse() ? pFileTemplate->GetRootNode() : nullptr;
-    if (!pRootNodeTemplate)
+    CXMLNode* templateRootNode = templateFile->GetRootNode();
+    if (!templateRootNode)
     {
-        CLogger::ErrorPrintf("Can't parse '%s'\n", *strTemplateFilename);
+        CLogger::ErrorPrintf("Template file '%s' has no root node\n", templateFileName.c_str());
         return false;
     }
 
     // Check that each item in the template also exists in the server config
-    bool      bChanged = false;
-    CXMLNode* pPrevNode = nullptr;
-    for (auto it = pRootNodeTemplate->ChildrenBegin(); it != pRootNodeTemplate->ChildrenEnd(); ++it)
-    {
-        CXMLNode* pNodeTemplate = *it;
-        SString   strNodeName = pNodeTemplate->GetTagName();
-        CXMLNode* pNode = m_pRootNode->FindSubNode(strNodeName);
-        if (!pNode)
+    bool configChanged = false;
+    CXMLNode* previousNode = nullptr;
+
+    for (auto it = templateRootNode->ChildrenBegin(); it != templateRootNode->ChildrenEnd(); ++it)
+    {
+        CXMLNode* templateNode = *it;
+        const std::string& templateNodeName = templateNode->GetTagName();
+
+        // Skip certain optional nodes
+        if (templateNodeName == "resource" || templateNodeName == "module")
+            continue;
+
+        CXMLNode* foundNode = m_pRootNode->FindSubNode(templateNodeName.c_str());
+        if (!foundNode)
         {
-            CLogger::LogPrintf("Adding missing '%s' to mtaserver.conf\n", *strNodeName);
-            SString strNodeValue = pNodeTemplate->GetTagContent();
-            SString strNodeComment = pNodeTemplate->GetCommentText();
-            pNode = m_pRootNode->CreateSubNode(strNodeName, pPrevNode);
-            pNode->SetTagContent(strNodeValue);
-            pNode->SetCommentText(strNodeComment, true);
-            bChanged = true;
+            const std::string templateNodeValue = templateNode->GetTagContent();
+            const std::string templateNodeComment = templateNode->GetCommentText();
```

Comment:
`GetCommentText` returns `SString`. This triggers an unnecessary string copying. It is recommended to use matching types for a variable and a function's return type.